Bugfix/problem with having more that 50 skus

brunovarelaAptoide opened the pull request:
**What does this PR do?**

   This PR solves the problem of asking the sku details with a request with more than 50 sku.

**Database changed?**

   No

**Where should the reviewer start?**

- AppcoinsBillingStubHelper.java

**How should this be manually tested?**

- ask for SKU details with no Wallet Installed.( Maybe add more 50 sku to the toolbox?? )

**What are the relevant tickets?**

  Tickets related to this pull-request: [APPC-1526](https://aptoide.atlassian.net/browse/APPC-1526)

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new libs, new keys, etc.) 
No



**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

brunovarelaAptoide committed:
changes

- Added support for more than 50 skus and pagination from the WS

brunovarelaAptoide committed:
changes

- Algorithm optimization to obtain skudetails

carvalho2707 commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AndroidBillingMapper.java:
```diff
@@ -145,8 +145,8 @@ public static LaunchBillingFlowResult mapBundleToHashMapGetIntent(Bundle bundle)
+  public static ArrayList<SkuDetails> mapSkuDetailsFromWS(String skuType, String skuDetailsresponse) {
+    ArrayList<SkuDetails> skuDetailsList = new ArrayList<SkuDetails>();
 
     if (skuDetailsresponse != "") {
```
Use "equals" to compare strings

carvalho2707 commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AndroidBillingMapper.java:
```diff
@@ -145,8 +145,8 @@ public static LaunchBillingFlowResult mapBundleToHashMapGetIntent(Bundle bundle)
 
-  public static SkuDetailsResult mapSkuDetailsFromWS(String skuType, String skuDetailsresponse) {
-    ArrayList<SkuDetails> arrayList = new ArrayList<SkuDetails>();
+  public static ArrayList<SkuDetails> mapSkuDetailsFromWS(String skuType, String skuDetailsresponse) {
```
skuDetailsresponse should have the R in Response capitalized

carvalho2707 commented on toolbox/src/main/java/com/asf/appcoins/toolbox/MainActivity.java:
```diff
@@ -125,9 +125,7 @@ public void onSkuDetailsButtonClicked(View view) {
-
     skuDetailsParams.setMoreItemSkus(skusList);
 
     Thread t = new Thread(new Runnable() {
```
new Runnable() and new SkuDetailsResponseListener() can be replaced with lambda expression:

 public void onSkuDetailsButtonClicked(View view) {
    SkuDetailsParams skuDetailsParams = new SkuDetailsParams();
    skuDetailsParams.setItemType(SkuType.inapp.toString());
    ArrayList<String> skusList = new ArrayList<>();
    skusList.add("gas");
    skuDetailsParams.setMoreItemSkus(skusList);

    Thread t = new Thread(() -> cab.querySkuDetailsAsync(skuDetailsParams,
        (responseCode, skuDetailsList) -> {
          Log.d(TAG, "responseCode: " + responseCode + "");
          for (SkuDetails sd : skuDetailsList) {
            Log.d(TAG, sd.toString());
          }
        }));

    t.start();
   }

carvalho2707 commented on toolbox/src/main/java/com/asf/appcoins/toolbox/MainActivity.java:
```diff
@@ -148,7 +146,6 @@ public void onSkuDetailsResponse(int responseCode, List<SkuDetails> skuDetailsLi
 
   public void makePaymentButtonClicked(View view) {
-
     Thread t = new Thread(new Runnable() {
```
same as above:


public void makePaymentButtonClicked(View view) {
    Thread t = new Thread(() -> {

      if (token != null) {
        cab.consumeAsync(token, (responseCode, purchaseToken) -> {
          Log.d(TAG, "consume response: "
              + responseCode
              + " "
              + "Consumed purchase with token: "
              + purchaseToken);
          token = null;
        });
      } else {
        Log.d(TAG, "Message: No purchase tokens available");
      }
    });
    t.start();
  }

carvalho2707 commented on appcoins-billing/src/main/java/com/appcoins/sdk/billing/GetSkuDetailsService.java:
```diff
@@ -24,11 +30,13 @@ public GetSkuDetailsService(final String serviceUrl, final String packageName,
-  public  String getSkuDetailsForPackageName() {
+  public String getSkuDetailsForPackageName() {
     String response = "";
     URL url = null;
```
Redudant initializer:   "= null" can be removed

carvalho2707 commented on appcoins-billing/src/main/java/com/appcoins/sdk/billing/GetSkuDetailsService.java:
```diff
@@ -1,17 +1,23 @@
 import java.net.URL;
+import java.util.ArrayList;
 import java.util.List;
+import org.json.JSONArray;
```
Removed unused imports

carvalho2707 commented on appcoins-billing/src/main/java/com/appcoins/sdk/billing/GetSkuDetailsService.java:
```diff
@@ -1,17 +1,23 @@
 public class GetSkuDetailsService {
 
   private final static String URL_PATH = "/inapp/8.20180518/packages/packageName/products?names=";
+  private static final String APPC = "APPC";
```
this constant is not used

carvalho2707 commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppcoinsBillingStubHelper.java:
```diff
@@ -73,17 +73,33 @@ public Bundle getSkuDetails(final int apiVersion, final String packageName, fina
-    SkuDetailsResult skuDetailsResult = AndroidBillingMapper.mapSkuDetailsFromWS(type, response);
+  private ArrayList<SkuDetails> getSkuDetailsFromService(List<String> sku, String packageName,
+      String type) {
+    ArrayList skuSendList = new ArrayList();
```
Unnecessary unchecked call. Replace with "List<String> skuSendList = new ArrayList<>();"

carvalho2707 reviewed (changes requested):
In Main Activity we have multiple Anonymous classes that can be replaced with lambda expressions:
Methods:

- [ ] onBuyGasButtonClicked
- [ ] onUpgradeAppButtonClicked
- [ ] onUpgradeAppButtonClicked
- [ ] makePaymentButtonClicked

brunovarelaAptoide committed:
Code review changes

- Removed unused variables
- Code optimization removeing anonymous classes.
- changed compare sign between strings to equals.
